timeutil/schedule.go

```diff
@@ -28,35 +29,49 @@ import (
 	"time"
 )
 
-var validTime = regexp.MustCompile(`^([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9])$`)
+// Match 0:00-24:00, where 24:00 means the later end of the day.
+var validTime = regexp.MustCompile(`^(([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9]))|(24:00)$`)
 
-type TimeOfDay struct {
+// Clock represents a hour:minute time within a day.
+type Clock struct {
 	Hour   int
 	Minute int
 }
 
-func (t TimeOfDay) String() string {
+func (t Clock) String() string {
 	return fmt.Sprintf("%02d:%02d", t.Hour, t.Minute)
 }
 
 // Sub subtracts `other` TimeOfDay from current and returns duration
```

Comment:
There are multiple occurrences of TimeOfDay still. Please grep throughout the whole code base to make sure we've moved it all over.